attempt to use the android sdk's cmake / gradle / ndk / ninja

gpx1000 opened the pull request:
blind test for CI.

gpx1000 committed:
see if Gerry's idea can be run directly in github's action for CI environment

gpx1000 committed:
update the package list for a future build and for now see if can get the build.yml to produce something that works.  We should probably also start removing things that are installed already and clean this up a bit; for now this is just a test.

gpx1000 committed:
use the ; instead of the : to id packages for sdk-manager

gpx1000 committed:
trying to install the ndk via the SDK manager doesn't work.  revert back to using the local.properties to ensure we are setting the ndk.  This is probably for the best anyway as no need to reinstall ndk.

gpx1000 committed:
set the local.properties after the build/android_gradle is completed.

ggfan commented on .github/workflows/build.yml:
```diff
@@ -60,16 +60,16 @@ jobs:
       - name: "Explicitly setup SDK path with local.properties."
         run: |
           echo "sdk.dir=/usr/local/android-sdk" > build/android_gradle/local.properties
           echo "ndk.dir=/usr/local/android-ndk" >> build/android_gradle/local.properties
```
maybe we do not have to do this to local.properties anymore: let us change the build.gradle to specify which version to use.

gpx1000 committed:
First attempt at cleaning up.
1.) remove the ndk being downloaded separately.  use the sdkmanager to retrieve it.
2.) run the ndk retrieval from dockerfile
3.) remove the sdk install steps from the ci build step just rely upon it from dockerfile.
4.) remove the generate android script to ensure that we don't get a lot of environment variables set that might not be needed -- probably need to revert this.
5.) use the ANDROID_HOME and ANDROID_SDK environment variables defined in Dockerfile.

gpx1000 committed:
readd the generate script to allow cmake create the project

gpx1000 committed:
fix the package-list.txt

gpx1000 committed:
Update build.yml

allow building in local fork.

gpx1000 committed:
use the docker image that is built locally.

gpx1000 committed:
Merge remote-tracking branch 'origin/update_Android_ci_build' into update_Android_ci_build

gpx1000 committed:
print out env

gpx1000 committed:
Update check.yml

gpx1000 committed:
try removing the explicit jnilibs directory setting

gpx1000 committed:
Merge remote-tracking branch 'origin/update_Android_ci_build' into update_Android_ci_build

gpx1000 committed:
try doing assemble debug instead of release

gpx1000 committed:
attempt to remove jnilibs

gpx1000 committed:
looks like CI can build with these

tomadamatkinson commented:
These changes were incorporated in #585
